### ISSUE TITLE
Midpoint Review - Accessibility Feedback - Decision Reviews Team, Board Appeal (Notice of Disagreement) VA Form 10182, Request an Extension Feature (OMB Approved No. 2900-0674)

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#76482]
## Next Steps for the VFS team

- [x] **Assign** this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [x] **Comment** on this ticket:
  - If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [x] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members`.
- [x] **Close the ticket** when all feedback has been addressed.

## Thoughts/questions

-

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform Governance reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- [x] **Must:** There are a few pages where there's some helper text that's likely to be missed by screen reader users. When a screen reader user is moving through a form flow like this, they typically don't have their screen reader announce everything on the page --- it's just too much and too repetitive to listen to. Instead, they enter "forms mode" for some screen readers, or otherwise just press [tab] to move between interactive elements.
   
   So when you tab to a form input, the screen reader will announce the label, the contents of the input, and any text that's associated with the input. The problem comes when you've got extra helper text in a `fieldset`, for example on the [Reason for extension page](https://staging.va.gov/decision-reviews/board-appeal/request-board-appeal-form-10182/extension-reason) which looks something like:
   
   ```
   <fieldset>
      <legend>
         <h3>Heading text</h3>
      </legend>
      
      <p>Some helper text</p>
      
      <div>
         <va-textarea label="Label text"></va-textarea>
      </div>
   </fieldset>
   ```
   
   When tabbing into the `va-textarea`, the screen reader will announce the `legend` text and the `textarea` label, but it won't announce the `p` text. It's technically not a WCAG failure because a screen reader user *could* hear the `p` text if they had their screen reader announce the whole page --- but they probably won't, which means information they might need could be accidentally hidden.
      
   There are a couple of possible solutions here, depending on what's being communicated:
   - Move the `p` text into the `hint` prop for the VA web component. The hint text is associated with the input using `aria-describedby` and will be announced.
   - Move the `p` into the `legend`:
      ```
      <legend>
         <h3>Heading text</h3>
         <p>Some helper text</p>
      </legend>
      ```
      You can style the helper text so it doesn't look like a legend and just looks like regular text. But for screen reader users, everything inside the `legend` will get announced when tabbing into the input --- so nothing gets lost.
   - For a non-code solution, see if there's a better way of phrasing the question so that the helper text isn't needed.
   
   No preference on my end for how you solve it, just so long as screen reader users hear everything they need so they can answer the questions.

## Governance team actions
- [x] Format feedback as individual tasks (check boxes)
- [x] Assign this ticket to the VFS team member that opened the Slack request
- [x] Add the VFS team product label
- [x] Add the VFS team the feature label (if applicable)
- [x] Add the touchpoint labels
- [x] Add the practice area labels
- [x] Add the Collaboration Cycle initiative milestone
- [x] Add the Collaboration Cycle initiative epic

**Comment by saderagsdale** [comment]
DK reached out to Brian. Waiting to hear back.

**Comment by davidakennedy** [comment]
I chatted with @briandeconinck about this some more. He'll list out anything else in this vein for the upcoming staging review.

I went back through this, and looked at the [screen mentioned](https://staging.va.gov/decision-reviews/board-appeal/request-board-appeal-form-10182/extension-reason) here:

```html
<fieldset class="vads-u-margin-y--2 rjsf-object-field">
  <legend class="schemaform-block-title" id="root__title">
    <h3 class="vads-u-margin-y--0">Reason for extension</h3>
  </legend>
  <p class="vads-u-margin-y--0">Tell us why you have good cause for an extension.</p>
  <div>
    <va-textarea class="rjsf-web-component-field hydrated" enableanalytics="false" hint="2300 characters max." label="Reason for requesting an extension:" maxlength="2300" name="root_extensionReason" required="true" uswds="true" value="" charcount="true"></va-textarea>
  </div>
</fieldset>
```

I don't think the helper text here needs to move because it's repeated in the label for the reason:

`Tell us why you have good cause for an extension.` vs. `Reason for requesting an extension:`

That said, we may be able to reduce that duplicative content.

I went back through the form and didn't notice any other helper text that was vital that needed to be moved into the legend.

**Comment by saderagsdale** [comment]
@briandeconinck Resolved in this ticket: #76909

**Comment by saderagsdale** [comment]
@davidakennedy I think we can close this, right?
